hw8 is done

Ltaudato1 opened the pull request.

Ltaudato1 committed:
hw8 is done

Ltaudato1 committed:
hw9

damir2407 commented on bot/src/main/java/edu/java/bot/KafkaConfiguration.java:
```diff
@@ -0,0 +1,25 @@
+@RequiredArgsConstructor
+public class KafkaConfiguration {
+
+    private final ApplicationConfig applicationConfig;
```
не используешь

damir2407 commented on bot/src/main/java/edu/java/bot/KafkaConfiguration.java:
```diff
@@ -0,0 +1,25 @@
+
+    @Bean
+    public NewTopic scrapperTopic() {
+        return TopicBuilder.name("${kafka.topic}")
```
> Все имена должны передаваться из конфигурационного файла, т.е. через ApplicationConfig.

По заданию

damir2407 replied:
у тебя же нет такого в конф файле

damir2407 commented on bot/src/main/java/edu/java/bot/LinkUpdateListener.java:
```diff
@@ -0,0 +1,33 @@
+
+    private final NotificationService notificationService;
+
+    @Autowired
```
Autowired лишний

damir2407 commented on bot/src/main/java/edu/java/bot/LinkUpdateListener.java:
```diff
@@ -0,0 +1,33 @@
+    @Autowired
+    private KafkaTemplate<String, String> kafkaTemplate;
+
+    @KafkaListener(topics = "${app.scrapper-topic.name}")
```
нет такой конфы в конф файле

damir2407 commented on bot/src/main/java/edu/java/bot/LinkUpdateListener.java:
```diff
@@ -0,0 +1,33 @@
+    @Bean
+    public KafkaListenerErrorHandler customErrorHandler() {
+        return (message, exception) -> {
+            kafkaTemplate.send("scrapper-topic_dlq", message.getPayload().toString());
```
снова же. Названия подхватывать из конфы нужно (по заданию)

damir2407 commented on scrapper/src/main/java/edu/java/configuration/ApplicationConfig.java:
```diff
@@ -11,7 +11,10 @@ public record ApplicationConfig(
+    AccessType databaseAccessType,
+
+    @NotNull
+    Boolean useQueue
```
notNull, который всегда null) У тебя нет настройки в конф файле

damir2407 commented on bot/src/main/java/edu/java/bot/KafkaConfiguration.java:
```diff
@@ -0,0 +1,25 @@
+@Configuration
+@EnableKafka
+@RequiredArgsConstructor
+public class KafkaConfiguration {
```
как создавал топики? В коде не нашел

кажется ты их не создавал

Ltaudato1 committed:
hw10

Ltaudato1 committed:
updated workflow

Ltaudato1 committed:
added jars

Ltaudato1 committed:
ACTUALLY added jars

Ltaudato1 committed:
updated docker-files

Ltaudato1 committed:
temporary fix

Ltaudato1 committed:
temporary fix v2

почему-то именно на гите докер билд вообще не проходит, хотя локально всё работает вообще отлично

Ltaudato1 committed:
Merge pull request #11 from Ltaudato1/hw10

hw10

Ltaudato1 committed:
Merge pull request #10 from Ltaudato1/hw9

hw9
